### DISCUSSION TITLE
Updating and correcting Windows Headers

### DISCUSSION THREAD

**Discussion by joncampbell123** [open-watcom/open-watcom-v2#1309] [General]
Hi @jmalak I gathered Windows API headers from the Windows 1.0 to 3.11 Windows For Workgroups headers from the WinworldPC website and posted them on GitHub for reference.

https://github.com/joncampbell123/windows_sdk_collection

I think it would be a great reference updating the Open Watcom Windows headers to reflect API versions and correct any mistakes. I still intend to make pull requests to update and improve the headers but this time I hope this reference allows anyone else here to also do the same.

Future additions to the SDK collection will eventually include Win32s, Windows 95/NT and later of course.

Anyone have any suggestions what to start with?

**Comment by jmalak** [comment]
I understand your effort, but be careful with regards to OpenWatcom because our headers (for 16-bit Windows) are modified for use with 32-bit Windows Extender only available for Windows 3.x. I am not sure if better solution is create new version for older version of Windows (1.0 and 2.0).

**Reply by joncampbell123** [reply]
It would be no different than the pull requests I made last December adding some missing #defines or #ifdefs to better match what the 1.0 and 2.0 SDKs do, and missing SDK functions as well. I'll keep the extender in mind. The Win386 extender can't provide functions it doesn't recognize, correct?

**Reply by jmalak** [reply]
Yes, there are some definition files which define exactly list of intercepted API entries etc. for Extender and in header files it is guarder by **`__WINDOWS_386__`** or **`__386__`** macro
I notice you only that there can be some odities in prototypes and data types etc. also in shared and 16-bit version.
Take into acount that most of prototypes is common for both version only some have both variants.
the lists of used functions is in bld\win386\conv\def directory where is single file per DLL.
